File: js/2018.js
const fs = require('fs');

const N = +fs.readFileSync('example.txt').toString().trim();

let start = 1;
let end = 1;
let value = 1;
let cnt = 0;

while (end <= N && start <= end) {
  console.log(start, end, value);
  if (value <= N) {
    if (value === N) {
      cnt++;
    }
    value += ++end;
  } else {
    value -= start++;
  }
}

console.log(cnt);

File: c++/2018.cpp
#include <iostream>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);
  int N;
  cin >> N;

  int start = 1;
  int end = 1;
  int sum = 1;
  int cnt = 0;

  while (end <= N && start <= end) {
    if (sum <= N) {
      if (sum == N) cnt++;
      sum += ++end;
    } else {
      sum -= start++;
    }
  }

  cout << cnt;

  return 0;
}